api/envoy/service/discovery/v3/discovery.proto

```diff
@@ -232,7 +263,59 @@ message DeltaDiscoveryResponse {
   config.core.v3.ControlPlane control_plane = 7;
 }
 
-// [#next-free-field: 8]
+// A set of dynamic parameter constraints associated with an individual
+// xDS resource. These constraints determine whether the resource matches
+// a subscription based on the set of dynamic parameters in the subscription.
+// This allows xDS servers and caching proxies to determine which variant of a
+// resource is appropriate for a given client.
+message DynamicParameterConstraints {
+  message Constraint {
+    // A list of one or more integer ranges.
+    // A value is considered to match if it falls in any of the ranges.
+    message IntegerRangeList {
+      // At least one of *min_value* or *max_value* must be set.
```

Comment:
Should you use WKT to make it clear that 0 is different from unset? Also can you clarify the behavior of someeone sets max < min or vice versa?